client/allocrunner/taskrunner/task_runner_hooks.go

```diff
@@ -211,7 +217,11 @@ func (tr *TaskRunner) prestart() error {
 		}
 
 		// Store the environment variables returned by the hook
-		tr.envBuilder.SetHookEnv(name, resp.Env)
+		if name == HookNameDevices {
```

Comment:
We should be passing the resp.Env here, not the origHookState, as it is, this causes a crash when origHookState is nil.